client/src/bundles/FieldEditor.js

```diff
@@ -2,107 +2,106 @@
+            thisLevel = ++depth;
+            break;
+          }
+          case 'EditableFieldGroupEnd': {
+            thisLevel = depth--;
+            break;
+          }
+          default: {
+            thisLevel = depth;
+          }
+        }
 
-				// Make sure gridfield buttons stick to top of page when user scrolls down
-				stickyHeaderInterval = setInterval(function () {
-					var offsetTop = fieldEditor.offset().top;
-					$buttonrow.width('100%');
-					if (offsetTop > navHeight || offsetTop === 0) {
-						$buttonrow.removeClass('stickyButtons');
-					} else {
-						$buttonrow.addClass('stickyButtons');
-					};
-				}, 300);
-			},
-			onunmatch: function () {
-				this._super();
+        $(this).toggleClass('infieldgroup', thisLevel > 0);
+        for (i = 1; i <= 5; i++) {
+          $(this).toggleClass(`infieldgroup-level-${i}`, thisLevel >= i);
+        }
+      });
 
-				clearInterval(stickyHeaderInterval);
-			}
-		});
+      // Make sure gridfield buttons stick to top of page when user scrolls down
+      stickyHeaderInterval = setInterval(() => {
+        const offsetTop = fieldEditor.offset().top;
+        $buttonrow.width('100%');
+        if (offsetTop > navHeight || offsetTop === 0) {
+          $buttonrow.removeClass('sticky-buttons');
+        } else {
+          $buttonrow.addClass('sticky-buttons');
+        }
+      }, 300);
+    },
+    onunmatch: () => {
+      this._super();
 
-		// When new fields are added..
-		$('.uf-field-editor .ss-gridfield-buttonrow .action').entwine({
-			onclick: function (e) {
-				this._super(e);
+      clearInterval(stickyHeaderInterval);
+    },
+  });
 
-				this.trigger('addnewinline');
-			}
-		});
+  // When new fields are added.
+  $('.uf-field-editor .ss-gridfield-buttonrow .action').entwine({
+    onclick: (e) => {
+      this._super(e);
 
-		$('.uf-field-editor').entwine({
-			onmatch: function () {
-				var self = this;
+      this.trigger('addnewinline');
+    },
+  });
 
-				this._super();
+  $('.uf-field-editor').entwine({
+    onmatch: () => {
+      this._super();
 
-				// When the 'Add field' button is clicked set a one time listener.
-				// When the GridField is reloaded focus on the newly added field.
-				this.on('addnewinline', function () {
-					self.one('reload', function () {
-						//If fieldgroup, focus on the start marker
-						var $newField = self.find('.ss-gridfield-item').last(), $groupEnd;
-						if ($newField.attr('data-class') === 'EditableFieldGroupEnd') {
-							$groupEnd = $newField;
-							$groupEnd.prev().find('.col-Title input').focus();
-							$newField = $groupEnd.add($groupEnd.prev());
-							$groupEnd.css('visibility', 'hidden');
-						} else {
-							$newField.find('.col-Title input').focus();
-						}
+      // When the 'Add field' button is clicked set a one time listener.
+      // When the GridField is reloaded focus on the newly added field.
+      this.on('addnewinline', () => {
+        this.one('reload', () => {
```

Comment:
This is the correct behaviour for fat arrow functions and es6 :) the transpiler will handle the correct `this` for you.

You can test it with adding `console.log(this)` before and after `this.on('addnewline', () => {` and even one after `this.one('reload', () => {`
it should all return the same `$('.uf-field-editor')`